central/resourcecollection/service/service_impl.go

```diff
@@ -132,13 +127,18 @@ func (s *serviceImpl) GetCollectionCount(ctx context.Context, request *v1.GetCol
 		return nil, errors.Errorf("%s env var is not enabled", features.ObjectCollections.EnvVar())
 	}
 
-	// parse query
-	parsedQuery, err := search.ParseQuery(request.GetQuery().GetQuery(), search.MatchAllIfEmpty())
-	if err != nil {
-		return nil, errors.Wrap(errox.InvalidArgs, err.Error())
+	var query *v1.Query
+	var err error
+
+	// query
+	if request.GetQuery() != nil {
```

Comment:
pull this out into a helper (see ListCollections)

Reply:
+1